File: 0/0/6/006.c
// Copyright (C) 2013 Jorge Aparicio

#include <stdio.h>

int main() {
  int i, sumOfSquares = 0, sumSquared = 0;

  for (i = 0; i < 101; i++) {
    sumSquared += i;
    sumOfSquares += i * i;
  }
  sumSquared *= sumSquared;

  printf("%d\n", sumSquared - sumOfSquares);

  return 0;
}

File: 0/0/6/006.rs
// Copyright (C) 2014 Jorge Aparicio

use std::iter::AdditiveIterator;
use std::num::pow;

fn main() {
    let sumSquared = pow(range(1, 101).sum(), 2);
    let sumOfSquares = range(1, 101).map(|x| pow(x, 2)).sum();

    println!("{}", sumSquared - sumOfSquares);
}

